registration/src/main/java/com/jpd/registration/service/StudentService.java

```
@@ -68,4 +70,17 @@ public Student updateStudent(Long id, StudentPayload payload)
 
         return studentRepo.save(student);
     }
+
+    public Map<String, Object> deleteStudent(Long id)
+    {
+        Student student = studentRepo.findById(id)
```

question: why do you need to find the student first?

---

"I chose to use findById() to ensure the student or school exists before deleting, instead of silently failing with deleteById()."